src/engraving/libmscore/undo.cpp

```diff
@@ -2132,13 +2141,23 @@ void AddExcerpt::redo(EditData*)
     excerpt->oscore()->addExcerpt(excerpt);
 }
 
+//---------------------------------------------------------
+//   RemoveExcerpt
+//---------------------------------------------------------
+
+RemoveExcerpt::RemoveExcerpt(Excerpt* ex)
+    : excerpt(ex)
+{
+    index = excerpt->oscore()->excerpts().indexOf(excerpt);
```

Comment:
It would be great to add a check on null here:

if (excerpt && execrpt->oscore()) {
index = excerpt->oscore()->excerpts().indexOf(excerpt);
}

Reply:
In general, none of the classes in `undo.cpp` have this kind of checks, it's the responsibility of the calling classes to make sure the arguments are correct. Which makes sense because these classes are used to do something via undo/redo instead of doing it straight away. In this case, the code come to the conclusion an `Excerpt` has to be removed but in stead of simply calling `masterScore->removeExcerpt(excerpt)` it uses `masterScore->undo(new RemoveExcerpt(excerpt)`. So sanity tests on arguments are expected to done **before** a new `RemoveExcerpt` object is created.